localstack-core/localstack/runtime/runtime.py

```diff
@@ -173,13 +173,21 @@ def create_from_environment() -> LocalstackRuntime:
     Creates a new runtime instance from the current environment. It uses a plugin manager to resolve the
     necessary components from the ``localstack.runtime.components`` plugin namespace to start the runtime.
 
-    TODO: perhaps we could control which components should be instantiated with a config variable/constant
-
     :return: a new LocalstackRuntime instance
     """
     hooks.on_runtime_create.run()
 
     plugin_manager = PluginManager(Components.namespace)
+    if config.RUNTIME_COMPONENTS:
+        try:
+            component = plugin_manager.load(config.RUNTIME_COMPONENTS)
+            return LocalstackRuntime(component)
+        except ValueError as e:
+            LOG.error(
+                "Could not load runtime component %s: %s. Falling back to loading all components.",
+                config.RUNTIME_COMPONENTS,
+                e,
+            )
```

Comment:
sorry i perhaps wasn't clear in the initial suggestions: i would actually raise an error here, something like:
i wouldn't silently fall back to something else if the user explicitly specifies something and expects it to load.

```suggestion
        except Exception as e:
            raise ValueError(
                "Could not load runtime components from config RUNTIME_COMPONENTS={config.RUNTIME_COMPONENTS}: {e}."
            ) from e
```